### PR TITLE
Tools: Be able to create a terminal in Mac OS

### PR DESCRIPTION
**Pull request by patrickelectric**
- Display env does not exist in the last version of High Sierra.
- SITL does not wait connection and fail, it was necessary to add a delay and a wait to connect properly. 

> Note: I would like to add a timeout in SITL to allow delays between the TCP connection when starting. Probably someone will be able to guide me. Removing 
74694b5

### CODE REVIEW

**Comment by OXINARF** [comment]
I haven't reviewed the PR yet, but addressing the SITL part: by default, SITL is supposed to not even start doing anything until you have to its first TCP port: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_SITL/UARTDriver.cpp#L274. If this isn't happening in macOS we have to figure out why.

**Comment by patrickelectric** [comment]
Thank you for pointing this @OXINARF, I'll take a look asap.

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -499,6 +499,8 @@ def run_in_terminal_window(autotest, name, cmd):
             progress("Cannot find %s process terminal" % name)
     else:
         p = subprocess.Popen(runme)
+        if under_macos():
```
Comment (OXINARF) [inline Tools/autotest/sim_vehicle.py]: If DISPLAY isn't needed, can't the DISPLAY check above be removed? Seems like that was intended to run under macOS.

**Comment by peterbarker** [comment]
@patrickelectric nudge!

**Comment by patrickelectric** [comment]
The intended behavior is already happening with the master branch.

**Pull request closed without merging**